### PR TITLE
Rust stable!

### PR DESCRIPTION
**Pull request by Nemikolh**
Fix neats and remove code that preventing the move to stable rust.

### CODE REVIEW

**Comment by ghost** [comment]
This PR does more than one things (: you did it again!

I will try as soon as I can… yet I imagine I need a Lycan server? Is there an offline mode?

**Comment by Nemikolh** [comment]
Yes there is! You can start the client with `sarosa --offline`. :)

File: sarosa_engine/src/events/actions.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
+use std::collections::HashMap;
```
Comment (Nemikolh) [inline sarosa_engine/src/events/actions.rs]: Okay, let me clean up this PR first. I shouldn't have committed this file.

**Comment by Nemikolh** [comment]
I think this is now easier to review. The other changes are minor and match a few changes that have also took place on lycan.

**Comment by ghost** [comment]
Okay, so now it comiles. However, I got some warnings one should fix eventually. If you do not do it, I will try. Nonetheless, this is not an issue to merge this PR. It’s a good opportunity to read your code.

```
   Compiling sarosa_engine v0.0.1 (file:///home/lethom/repos/greenpix/sarosa)
sarosa_engine/src/net/real/reader.rs:38:65: 38:67 warning: unused variable: `pv`, #[warn(unused_variables)] on by default
sarosa_engine/src/net/real/reader.rs:38             NewEntity { entity, position: Vec2d { x, y }, skin, pv } => {
                                                                                                        ^~
sarosa_engine/src/net/real/reader.rs:66:49: 66:51 warning: unused variable: `pv`, #[warn(unused_variables)] on by default
sarosa_engine/src/net/real/reader.rs:66             Position { entity, position, speed, pv } => {
                                                                                        ^~
sarosa_engine/src/net/real/reader.rs:28:75: 28:86 warning: private type in public interface, #[warn(private_in_public)] on by default
sarosa_engine/src/net/real/reader.rs:28     pub fn consume_event(&mut self, server_event: Notification) -> Option<ServerEvent> {
                                                                                                                  ^~~~~~~~~~~
sarosa_engine/src/net/real/reader.rs:28:75: 28:86 warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
sarosa_engine/src/net/real/reader.rs:28:75: 28:86 note: for more information, see the explanation for E0446 (`--explain E0446`)
sarosa_engine/src/net/real/mod.rs:110:59: 110:70 warning: private type in public interface, #[warn(private_in_public)] on by default
sarosa_engine/src/net/real/mod.rs:110     pub fn start_reader_thread(&mut self, tx_serv: Sender<ServerEvent>, rx_error_reader: Receiver<()>) {
                                                                                                ^~~~~~~~~~~
sarosa_engine/src/net/real/mod.rs:110:59: 110:70 warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
sarosa_engine/src/net/real/mod.rs:110:59: 110:70 note: for more information, see the explanation for E0446 (`--explain E0446`)
sarosa_engine/src/net/fake.rs:117:59: 117:70 warning: private type in public interface, #[warn(private_in_public)] on by default
sarosa_engine/src/net/fake.rs:117     pub fn start_reader_thread(&mut self, tx_serv: Sender<ServerEvent>, _: Receiver<()>) {
                                                                                            ^~~~~~~~~~~
sarosa_engine/src/net/fake.rs:117:59: 117:70 warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
sarosa_engine/src/net/fake.rs:117:59: 117:70 note: for more information, see the explanation for E0446 (`--explain E0446`)
   Compiling sarosa v0.0.1 (file:///home/lethom/repos/greenpix/sarosa)

```

**Comment by Nemikolh** [comment]
This PR fixes #3.

**Comment by Nemikolh** [comment]
I think I can fix easily the unused variable ones but not the private in public. This one looks more like a bug to me because it doesn't make a lot of sense. Still, I guess there's a workaround to get the privacy I'd like without exposing the type.

In the error generated, Rust complains about the fact that the type is private but if you look at the code the public one is less visible than the private one.

**Comment by ghost** [comment]
I tried, it worked, so I merge.

**Pull request merged**